common/policy_plugin.go

```diff
@@ -0,0 +1,252 @@
+	return &PolicyStoreServer{Impl: p.Impl}, nil
+}
+
+func (p *PolicyStorePlugin) Client(b *plugin.MuxBroker, c *rpc.Client) (interface{}, error) {
+	return &PolicyStoreRPC{client: c}, nil
+}
+
+type GetPolicyArgs struct {
+	TenantId    int
+	TokenFormat TokenFormat
+}
+
+type PutPolicyArgs struct {
+	TenantId int
+	Policy   *Policy
+}
+
+type PolicyStoreServer struct {
+	Impl IPolicyStore
+}
+
+func (s *PolicyStoreServer) GetName(args interface{}, resp *string) error {
+	*resp = s.Impl.GetName()
+	return nil
+}
+
+func (s *PolicyStoreServer) Init(args PolicyStoreParams, resp *string) error {
+	return s.Impl.Init(args)
+}
+
+func (s *PolicyStoreServer) GetPolicy(args GetPolicyArgs, resp *Policy) error {
+	policy, err := s.Impl.GetPolicy(args.TenantId, args.TokenFormat)
+	*resp = *policy
+	return err
+}
+
+func (s *PolicyStoreServer) PutPolicy(args PutPolicyArgs, resp *interface{}) error {
+	return s.Impl.PutPolicy(args.TenantId, args.Policy)
+}
+
+func (s *PolicyStoreServer) Close(args interface{}, resp *interface{}) error {
+	return s.Impl.Close()
+}
+
+type PolicyStoreRPC struct {
+	client *rpc.Client
+}
+
+func (e *PolicyStoreRPC) GetName() string {
+	var resp string
+	err := e.client.Call("Plugin.GetName", new(interface{}), &resp)
+	if err != nil {
+		log.Printf("ERROR during GetName RPC: %v\n", err)
+		return ""
+	}
+	return resp
+}
+
+func (e *PolicyStoreRPC) Init(args PolicyStoreParams) error {
+	return e.client.Call("Plugin.Init", args, nil)
+}
+
+func (e *PolicyStoreRPC) GetPolicy(tenantId int, tokenFormat TokenFormat) (*Policy, error) {
+	var resp Policy
+
+	args := GetPolicyArgs{TenantId: tenantId, TokenFormat: tokenFormat}
+	err := e.client.Call("Plugin.GetPolicy", args, &resp)
+
+	return &resp, err
+}
+
+func (e *PolicyStoreRPC) PutPolicy(tenantId int, policy *Policy) error {
+	args := PutPolicyArgs{TenantId: tenantId, Policy: policy}
+	// TODO: figure out why the last argument here must be non-nil
+	return e.client.Call("Plugin.PutPolicy", args, new(interface{}))
+}
+
+func (e *PolicyStoreRPC) Close() error {
+	return e.client.Call("Plugin.Close", new(interface{}), nil)
+}
+
+type PolicyEnginePlugin struct {
+	Impl IPolicyEngine
+}
+
+func (p *PolicyEnginePlugin) Server(*plugin.MuxBroker) (interface{}, error) {
+	return &PolicyEngineServer{Impl: p.Impl}, nil
+}
+
+func (p *PolicyEnginePlugin) Client(b *plugin.MuxBroker, c *rpc.Client) (interface{}, error) {
+	return &PolicyEngineRPC{client: c}, nil
+}
+
+type PolicyEngineArgs struct {
+	Evidence     map[string]interface{}
+	Endorsements map[string]interface{}
+}
+
+type GetAttetationResultArgs struct {
+	PolicyEngineArgs
+	Simple bool
+}
+
+type PolicyEngineServer struct {
+	Impl IPolicyEngine
+}
+
+func (s *PolicyEngineServer) GetName(args interface{}, resp *string) error {
+	*resp = s.Impl.GetName()
+	return nil
+}
+
+func (s *PolicyEngineServer) Init(args PolicyEngineParams, resp *interface{}) error {
+	return s.Impl.Init(args)
+}
+
```

Comment:
Might be a good idea for Functions beneath Line 134 to have a one liner comment, as to from where they fetch the set, like Policy, Claims or Attestation Results.

Reply:
This is documented as part of `PolicyEngine` interface, which this is just forwarding. This is also part of the internal plugin implementation and is not intended to be use by anyone else.

Reply:
Ok, Thanks then please do resolve the comment!